### ISSUE TITLE
[250522_Improve EffiSegNet] EffiSegNet 모델 개선 및 학습

### ISSUE THREAD

**Issue by WannaBeSuperteur** [WannaBeSuperteur/AI_Projects#13]
## Project README

- [Project README](https://github.com/WannaBeSuperteur/AI_Projects/blob/main/2025_05_22_Improve_EffiSegNet/README.md)
- [Schedule Table](https://github.com/WannaBeSuperteur/AI_Projects/blob/main/2025_05_22_Improve_EffiSegNet/README.md#3-%ED%94%84%EB%A1%9C%EC%A0%9D%ED%8A%B8-%EC%9D%BC%EC%A0%95)

## Task List

- [x] 모델 개선 방안 도출 (05.24 토, 1d)
  - https://github.com/WannaBeSuperteur/AI_Projects/issues/12
- [ ] 모델 개선 (05.24 토, 1d)
  - [ ] ElasticTransform 제거 (05.24 토, 1d)
    - 근거: [demo](https://huggingface.co/spaces/qubvel-hf/albumentations-demo) 에서 테스트 결과, 해당 Transformation 을 **논문 코드의 옵션에 따른 arguments 를 이용하여** 적용 시 이미지가 미 적용 시 이미지와 차이 없음
  - [ ] ColorJitter, Affine Transformation 확률 0.5 → 0.8 로 증가 (05.24 토, 1d)
  - [ ] 5개의 'level' (feature map 의 위치) 각각에 대해 학습 가능한 가중치를 이용한 gating 메커니즘 적용 (05.24 토, 1d)
  - [ ] 검은색 박스를 랜덤 사이즈로 랜덤한 귀퉁이에 하나 더 추가하는 Augmentation 추가 (05.24 토, 1d)
    - 단, 해당 부분을 mask 에서 항상 background 로 처리해야 함
    - 해당 부분은 **Kvasir-SEG 데이터셋에 특화** 된 Augmentation 임
- [ ] 개선된 모델 학습 (05.24 토 - 05.25 일, 2d)

**Comment by WannaBeSuperteur** [comment]
## Original EffiSegNet Model

|구분|Dice score|IoU score|recall|precision|
|---|---|---|---|---|
|valid - last epoch|0.8667|0.8011|0.8578|0.8941|
|test|**0.9445**|**0.8980**|**0.9368**|**0.9647**|
|average|0.9056|0.8496|0.8973|0.9294|

**Comment by WannaBeSuperteur** [comment]
## EffiSegNet Model Modified

* ElasticTransform 제거
* ColorJitter, Affine Transformation 확률 0.5 → 0.8 로 증가
* 비교는 **Original EffiSegNet Model** 대비

|구분|Dice score|IoU score|recall|precision|
|---|---|---|---|---|
|test|**0.9406**<br>(▼ 0.0039)|**0.8913**<br>(▼ 0.0067)|**0.9259**<br>(▼ 0.0109)|**0.9626**<br>(▼ 0.0021)|

**Comment by WannaBeSuperteur** [comment]
## EffiSegNet Model Modified (2차 / 05.24 12:30)

* **Original EffiSegNet Model** 과 비교한 변경점
  * ColorJitter, Affine Transformation 확률 0.5 → 0.8 로 증가 **(ElasticTransform 재 적용)**
* 비교는 **Original EffiSegNet Model** 대비

|구분|Dice score|IoU score|recall|precision|
|---|---|---|---|---|
|test|**0.9363**<br>(▼ 0.0082)|**0.8860**<br>(▼ 0.0120)|**0.9295**<br>(▼ 0.0073)|**0.9587**<br>(▼ 0.0060)|

* 추정 원인
  * [상기 코멘트](https://github.com/WannaBeSuperteur/AI_Projects/issues/13#issuecomment-2906270135) 와 동일

**Comment by WannaBeSuperteur** [comment]
## EffiSegNet Model Modified (3차 / 05.24 14:15)

* **Original EffiSegNet Model** 과 비교한 변경점
  * ElasticTransformation 제거 **(ColorJitter, Affine Transformation 확률은 각각 0.5 로 유지)**
* 비교는 **Original EffiSegNet Model** 대비

|구분|Dice score|IoU score|recall|precision|
|---|---|---|---|---|
|test|**0.9386**<br>(▼ 0.0059)|**0.8904**<br>(▼ 0.0076)|**0.9389**<br>(▲ 0.0021)|**0.9559**<br>(▼ 0.0088)|

* 추정 원인
  * 다음 2가지 옵션 조정 (다른 모든 옵션은 그대로 유지) 결과, Test Dice Score 가 **향상되는 경우와 하락하는 경우** 가 아래 그림과 같이 **모두 각각 1가지씩 존재**
    * ColorJitter, Affine Transformation 확률 0.5 → 0.8 로 증가
    * ElasticTransformation 제거
  * 따라서, 위 2가지 옵션은 모두 **성능에 큰 영향이 없고**, 실험을 통해 나타난 약간의 성능 변동은 오차 범위 이내의 noise 성 변동으로 추정

![Image](https://github.com/user-attachments/assets/20992f11-8e56-4578-884f-0f2ea1172d38)

**Comment by WannaBeSuperteur** [comment]
## EffiSegNet Model Modified (4차 / 05.24 16:00)

* **Original EffiSegNet Model** 과 비교한 변경점
  * ColorJitter augmentation 약화 (brightness ```[0.6, 1.6]``` → ```[0.8, 1.4]```, contrast ```0.2``` → ```0.15```, saturation ```0.1``` → ```0.075```)
  * ElasticTransform 제거
  * ColorJitter, Affine Transformation 확률 0.5 → 0.8 로 증가
* 비교는 **Original EffiSegNet Model** 대비

|구분|Dice score|IoU score|recall|precision|
|---|---|---|---|---|
|test|**0.9421**<br>(▼ 0.0024)|**0.8944**<br>(▼ 0.0036)|**0.9385**<br>(▲ 0.0017)|**0.9590**<br>(▼ 0.0057)|

**Comment by WannaBeSuperteur** [comment]
## EffiSegNet Model Modified (6차 / 05.25 12:30)

* **Original EffiSegNet Model** 과 비교한 변경점
  * sigmoid 함수 적용 전의 각 픽셀 별 logit 값에 대한 segmentation threshold 를 0.0 → 0.5 로 수정
  * **logit - mask 간 BCE loss** 계산 시, mask 값이 FALSE → 0.0, TRUE → 1.0 으로 mapping 되어 있음을 고려
* 비교는 **Original EffiSegNet Model** 대비

|구분|Dice score|IoU score|recall|precision|
|---|---|---|---|---|
|test|**0.9304**<br>(▼ 0.0006)|**0.8790**<br>(▼ 0.0013)|**0.9295**<br>(▼ 0.0068)|**0.9594**<br>(▲ 0.0056)|

**Comment by WannaBeSuperteur** [comment]
## EffiSegNet Model Modified (7차 / 05.25 16:30)

* **Original EffiSegNet Model** 과 비교한 변경점
  * segmentation result map (threshold 적용 전) 에서, 가로, 세로, 대각선으로 4칸 떨어진 픽셀 간의 값의 차이를 loss term 으로 추가 (즉, 기존 Dice + Cross Entropy Loss 에 해당 loss term 을 추가로 더함) 
* 비교는 **Original EffiSegNet Model** 대비

|구분|Dice score|IoU score|recall|precision|
|---|---|---|---|---|
|test|**0.9347**<br>(▲ 0.0037)|**0.8824**<br>(▲ 0.0021)|**0.9528**<br>(▲ 0.0165)|**0.9409**<br>(▼ 0.0129)|

**Comment by WannaBeSuperteur** [comment]
## EffiSegNet Model Modified (8차 / 05.25 22:45)

* **Original EffiSegNet Model** 과 비교한 변경점
  * segmentation result map (threshold 적용 전) 에서, 가로, 세로, 대각선으로 4칸 떨어진 픽셀 간의 값의 차이를 loss term 으로 추가 (즉, 기존 Dice + Cross Entropy Loss 에 해당 loss term 을 추가로 더함) 
* 비교는 **Original EffiSegNet Model** 대비

|구분|Dice score|IoU score|recall|precision|
|---|---|---|---|---|
|test|**0.9315**<br>(▲ 0.0005)|**0.8772**<br>(▼ 0.0031)|**0.9501**<br>(▲ 0.0138)|**0.9366**<br>(▼ 0.0172)|
